src/components/organisms/Settings/Project/Dataset/hooks.ts

```diff
@@ -25,33 +26,55 @@ export default (projectId: string) => {
 
   const sceneId = sceneData?.scene?.id;
 
-  const { data, refetch } = useDatasetSchemasQuery({
+  const { data } = useDatasetSchemasQuery({
     variables: { projectId: projectId ?? "" },
     skip: !projectId,
   });
 
   const nodes = data?.scene?.datasetSchemas.nodes ?? [];
 
   const datasetSchemas = nodes.filter(Boolean) as DatasetSchemas;
+  const client = useApolloClient();
 
-  const [importDatasetMutation] = useImportDatasetMutation();
-
-  const importDataset = useCallback(
-    (file: FileList) => {
-      sceneId && importDatasetMutation({ variables: { file, sceneId } });
+  const [removeDatasetSchema] = useRemoveDatasetMutation();
+  const handleRemoveDataset = useCallback(
+    async (schemaId: string) => {
+      await removeDatasetSchema({
+        variables: {
+          schemaId,
+          force: true,
+        },
+      });
+      // re-render
+      await client.resetStore();
     },
-    [sceneId, importDatasetMutation],
+    [client, removeDatasetSchema],
   );
 
-  const [removeDatasetSchemaMutation] = useRemoveDatasetMutation();
+  // Add
+  const [importData] = useImportDatasetMutation();
 
-  const removeDatasetSchema = useCallback(
-    async (schemaId: string) => {
-      await removeDatasetSchemaMutation({ variables: { schemaId } });
-      await refetch();
+  const handleDatasetImport = useCallback(
+    async (file: File, schemeId: string | null) => {
+      if (!sceneId) return;
+      await importData({
+        variables: {
+          file,
+          sceneId,
+          datasetSchemaId: schemeId,
+        },
+      });
+      // re-render
+      await client.resetStore();
```

Comment:
It's good this time, but I think it's better to purge the cache partially if possible.